File: backend/product_service/src/main/java/ucv/app_inventory/application/services/ProductService.java
package ucv.app_inventory.application.services;

import org.springframework.data.domain.Page;
import ucv.app_inventory.application.DTO.SupplierDTO;
import ucv.app_inventory.domain.entities.Product;

import java.util.List;

public interface ProductService {

    Page<Product> listProducts(int page, int size);

    Product createProduct(final Product product);

    Product updateProduct(final Product product);

    void deleteProduct(Product product);

    Product findProductById(final Long id);

    Page<Product> findProductsByName(String name, int page, int size);

    Page<Product> findProductsByStatus(Product.Status status, int page, int size);

    Page<Product> findProductsByCategoryName(String categoryName, int page, int size);

    SupplierDTO getSupplierDetails(Long supplierId);

    List<SupplierDTO> getSuppliersForProduct(Long productId);


}

File: frontend/src/services/products/ProductService.js
// services/products/ProductService.js
import config from '../../config';
import ApiProduct from './apiProduct';

const API_PRODUCTS = config.API_PRODUCTS;

const ProductService = {
    listProducts: (page = 0, size = 15) =>
        ApiProduct(`${API_PRODUCTS}/listProducts?page=${page}&size=${size}`),

    saveProduct: (product) =>
        ApiProduct(`${API_PRODUCTS}/saveProduct`, 'POST', product),

    updateProduct: (id, product) =>
        ApiProduct(`${API_PRODUCTS}/updateProduct/${id}`, 'PUT', product),

    deleteProduct: (id) =>
        ApiProduct(`${API_PRODUCTS}/deleteProduct/${id}`, 'DELETE'),

    findProductById: (id) =>
        ApiProduct(`${API_PRODUCTS}/findProductById/${id}`),

    findByStatus: (status, page = 0, size = 15) =>
        ApiProduct(`${API_PRODUCTS}/findByStatus?status=${status}&page=${page}&size=${size}`),

    findByCategoryName: (categoryName, page = 0, size = 15) =>
        ApiProduct(`${API_PRODUCTS}/findByCategoryName?categoryName=${categoryName}&page=${page}&size=${size}`),

    findByName: (name, page = 0, size = 15) =>
        ApiProduct(`${API_PRODUCTS}/findByName?name=${name}&page=${page}&size=${size}`),
};

export default ProductService;
